File: modules/physical-triggers/src/PhysicalTriggersModule.ts
import { NativeModule, requireNativeModule } from 'expo';
import type { VolumeTriggerConfig, ShakeTriggerConfig } from './PhysicalTriggers.types';

declare class PhysicalTriggersModule extends NativeModule {
  configureVolumeTrigger(config: VolumeTriggerConfig): Promise<boolean>;
  configureShakeTrigger(config: ShakeTriggerConfig): Promise<boolean>;
  startShakeTest(config: { highPassAlpha: number; jerkThreshold: number }): Promise<boolean>;
  stopShakeTest(): Promise<boolean>;
  simulateVolumePress(): Promise<boolean>;
  simulateShake(jerk?: number): Promise<boolean>;
  isVolumeTriggerEnabled(): Promise<boolean>;
  isShakeTriggerEnabled(): Promise<boolean>;
  addListener(eventName: string, listener: (event: any) => void): { remove: () => void };
  removeListeners(count: number): void;
}

export default requireNativeModule<PhysicalTriggersModule>('PhysicalTriggers');

File: modules/physical-triggers/ios/PhysicalTriggersModule.swift
import ExpoModulesCore

public class PhysicalTriggersModule: Module {
  public func definition() -> ModuleDefinition {
    Name("PhysicalTriggers")

    Events("onPanicTrigger", "onShakeTestSample")

    AsyncFunction("configureVolumeTrigger") { (config: [String: Any]) -> Bool in
      return true
    }

    AsyncFunction("configureShakeTrigger") { (config: [String: Any]) -> Bool in
      return true
    }

    AsyncFunction("startShakeTest") { (config: [String: Any]) -> Bool in
      return true
    }

    AsyncFunction("stopShakeTest") { () -> Bool in
      return true
    }

    AsyncFunction("simulateVolumePress") { () -> Bool in
      return true
    }

    AsyncFunction("simulateShake") { (jerk: Double?) -> Bool in
      return true
    }

    AsyncFunction("isVolumeTriggerEnabled") { () -> Bool in
      return false
    }

    AsyncFunction("isShakeTriggerEnabled") { () -> Bool in
      return false
    }
  }
}
